InnerEye/ML/dataset/full_image_dataset.py

```diff
@@ -262,6 +264,42 @@ def _load_dataset_sources(self) -> Dict[str, PatientDatasetSource]:
                                     )
 
 
+def converts_channels_to_file_paths(channels: List[str],
+                          rows: pd.DataFrame,
+                          local_dataset_root_folder: Path,
+                          patient_id: str) -> List[Path]:
+    """
+    Returns: 1) The full path for files specified in the training, validation and testing datasets, and
+             2) Missing channels or missing files.
+
+    :param channels: channel type defined in the configuration file
+    :param rows: Input Pandas dataframe object containing subjectIds, path of local dataset, channel information
+    :param local_dataset_root_folder: Root directory which points to the local dataset
+    :param patient_id: string which contains subject identifier
+    """
+    paths: List[Path] = []
+    failed_channel_info: str = ''
+
+    for channel_id in channels:
+        row = rows.loc[rows[CSV_CHANNEL_HEADER] == channel_id]
+        channel_failure_flag: bool = False
```

Comment:
I think you can remove this flag, and replace it with a check for failed_channel_info being empty

Reply:
Removed flag and added "else" condition. In general can't check check for failed_channel_info being empty since accumulates errors from multiple channels for same subject.